common/integration.go

```diff
@@ -0,0 +1,68 @@
+package common
+
+import (
+	"fmt"
+
+	config "github.com/de1ux/prowler/config/v1"
+	services "github.com/de1ux/prowler/services/v1"
+	vcs "github.com/de1ux/prowler/vcs/v1"
+
+	"github.com/de1ux/prowler/services/v1/bamboo"
+	"github.com/de1ux/prowler/vcs/v1/bitbucket"
+)
+
+type integration func(config *config.Config) (vcs.Client, []services.Client, error)
+
+// integrations combines vcs/services specific configurations into versioned generic vcs/services clients
+var integrations = map[string]integration{
+	"bitbucket_and_bamboo": func(config *config.Config) (vcs.Client, []services.Client, error) {
```

Comment:
The original idea was to make this take a `config.Config` and produce a Manifest. The thinking behind this is that it lets VCS/Services be created and invoked in ways that don't follow the flow
```
GetPRs -> GetServiceStates -> GetManifest
```

I.e. when a GetPrs operation isn't available for a VCS, the flow may need to be inverted

```
GetServiceStates -> GetPrsByServiceState -> GetManifest
```

I didn't capture this idea cleanly here. I'm open to discussion if this design sounds rediculous